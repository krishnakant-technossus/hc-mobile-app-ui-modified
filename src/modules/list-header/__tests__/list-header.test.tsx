// npx jest ./src/modules/list-header/__tests__/list-header.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import ListHeader from '../list-header';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('ListHeader', () => {
  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <ListHeader title="Categories" />
        </ThemeProvider>,
      );

      const title = wrapper.getByTestId('title');

      expect(title).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders link label correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <ListHeader title="Categories" linkLabel="View All" />
        </ThemeProvider>,
      );

      const linkLabel = wrapper.getByText('View All');

      expect(linkLabel).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call the onLinkPress method when link is pressed', () => {
      const onLinkPress = jest.fn();

      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <ListHeader
            title="Categories"
            onLinkPress={onLinkPress}
            linkLabel="View All"
          />
        </ThemeProvider>,
      );

      const link = wrapper.getByTestId('link');

      fireEvent.press(link);

      expect(onLinkPress).toHaveBeenCalledTimes(1);
    });
  });
});
