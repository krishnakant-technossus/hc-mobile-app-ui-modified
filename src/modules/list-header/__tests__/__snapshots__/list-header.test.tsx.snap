// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1C2530",
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "700",
          "textAlign": "left",
        },
      ]
    }
    testID="title"
    variant="h1"
  >
    Categories
  </Text>
</View>
`;

exports[`ListHeader Rendering renders link label correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1C2530",
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "700",
          "textAlign": "left",
        },
      ]
    }
    testID="title"
    variant="h1"
  >
    Categories
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="link"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#00AEFF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
            "textAlign": "left",
          },
        ]
      }
      testID="link-label"
      variant="h6"
    >
      View All
    </Text>
  </View>
</View>
`;
